paimon-format/src/main/java/org/apache/paimon/format/orc/OrcReaderFactory.java

```diff
@@ -183,7 +183,7 @@ private RecordIterator<InternalRow> convertAndGetIterator(
             int batchSize = orcBatch.size;
             paimonColumnBatch.setNumRows(batchSize);
             result.reset(batchSize);
-            result.resetRowPosition(rowNumber);
```

Comment:
Just keep it as it is.
We can implement `ColumnarRowIterator.returnedPosition` to `nextGlobalPosition - 1`?